pkg/xray/resource_xray_licenses_report.go

```diff
@@ -0,0 +1,91 @@
+package xray
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+func resourceXrayLicensesReport() *schema.Resource {
+	var licensesFilterSchema = map[string]*schema.Schema{
+		"component": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      "Artifact's component.",
+		},
+		"artifact": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      "Artifact name.",
+		},
+		"unknown": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Unknown displays the components that Xray could not discover any licenses for.",
+		},
+		"unrecognized": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Unrecognized displays the components that Xray found licenses for, but these licenses are not Xray recognized licenses.",
+		},
+		"license_names": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Set:         schema.HashString,
+			Optional:    true,
+			Description: "Filter licenses by names. Only one of 'license_names' or 'license_patterns' can be set.",
+		},
+		"license_patterns": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Set:         schema.HashString,
+			Optional:    true,
+			Description: "Filter licenses by patterns. Only one of 'license_names' or 'license_patterns' can be set.",
+		},
+		"scan_date": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			MaxItems:    1,
+			Description: "",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"start": {
+						Type:             schema.TypeString,
+						Optional:         true,
+						ValidateDiagFunc: validator.StringIsNotEmpty,
+						Description:      "Scan start date.",
+					},
+					"end": {
+						Type:             schema.TypeString,
+						Optional:         true,
+						ValidateDiagFunc: validator.StringIsNotEmpty,
```

Comment:
We should apply validation to date/time string: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation#IsRFC3339Time

Reply:
Replaced with `ValidateDiagFunc: validation.ToDiagFunc(validation.IsRFC3339Time),` for all date attributes.